core/connection/manager.go

```diff
@@ -282,16 +285,35 @@ func (manager *connectionManager) checkSessionIP(dialog communication.Dialog, se
 }
 
 // sendSessionStatus sends session connectivity status to other peer.
-func (manager *connectionManager) sendSessionStatus(dialog communication.Dialog, sessionID session.ID, code connectivity.StatusCode, errDetails error) error {
+func (manager *connectionManager) sendSessionStatus(dialog communication.Dialog, channel p2p.Channel, consumerID identity.Identity, sessionID session.ID, code connectivity.StatusCode, errDetails error) error {
 	var errDetailsMsg string
 	if errDetails != nil {
 		errDetailsMsg = errDetails.Error()
 	}
-	return manager.connectivityStatusSender.Send(dialog, &connectivity.StatusMessage{
+
+	if channel == nil {
+		return manager.connectivityStatusSender.Send(dialog, &connectivity.StatusMessage{
+			SessionID:  string(sessionID),
+			StatusCode: code,
+			Message:    errDetailsMsg,
+		})
+	}
+
+	sessionStatus := &pb.SessionStatus{
+		ConsumerID: consumerID.Address,
 		SessionID:  string(sessionID),
-		StatusCode: code,
+		Code:       uint32(code),
 		Message:    errDetailsMsg,
-	})
+	}
+
+	log.Debug().Msgf("Sending session status P2P message to %q: %s", p2p.TopicSessionCreate, sessionStatus.String())
```

Comment:
In log message it should be p2p.TopicSessionStatus too.

Reply:
Fixed it. Thanks.